tests/src/testcommon.hh

```diff
@@ -187,139 +186,88 @@ template <typename NonLinearOperator>
+  Eigen::MatrixXd computedStress(nodes, dim * (dim + 1) / 2);
+  computedStress.setZero();
 
-  using namespace Ikarus;
-  auto gridView = fe.localView().globalBasis().gridView();
-  auto element  = elements(gridView).begin();
+  auto& displacement = nonLinearOperator.firstParameter();
 
+  if (fe.localView().element().type() == Dune::GeometryTypes::simplex(FiniteElement::Traits::mydim)) {
+    Eigen::Matrix<double, 1, dim*(dim + 1) / 2> constantStresses;
+    if constexpr (dim == 2) {
+      displacement << 0, 0, 2, 0, 1, 0;
+      constantStresses << 2197.80219780, 659.34065934, 384.61538462;
+      expectedStress << constantStresses, constantStresses, constantStresses;
+    } else {
+      displacement << 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0;
+      constantStresses << 576.92307692, 1346.15384615, 576.92307692, 0, 384.61538462, 769.23076923;
+      expectedStress << constantStresses, constantStresses, constantStresses, constantStresses;
+    }
+  } else if (fe.localView().element().type() == Dune::GeometryTypes::cube(FiniteElement::Traits::mydim)) {
+    if constexpr (dim == 2) {
+      displacement << 0, 0, 1, 1, 1, 1, 1, 1;
+      if constexpr (requires { fe.setEASType(4); }) {
+        fe.setEASType(4);
+        expectedStress << 1214.28571429, 1214.28571429, 384.61538462, 1214.28571429, 214.28571429, 384.61538462,
+            214.28571429, 1214.28571429, 384.61538462, 214.28571429, 214.28571429, 384.61538462;
+      } else {
+        expectedStress << 1428.57142857, 1428.57142857, 769.23076923, 1098.90109890, 329.67032967, 384.61538462,
+            329.67032967, 1098.90109890, 384.61538462, 0, 0, 0;
+      }
+    } else {
+      displacement << 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0;
+      expectedStress << 576.92307692, 1346.15384615, 576.92307692, 384.61538462, 0, 384.61538462, 0, 0, 0, 0, 0, 0,
+          -1346.15384615, 192.30769231, -1346.15384615, 0, -769.23076923, 0, -1346.15384615, -576.92307692,
+          -576.92307692, 0, -384.61538462, -384.61538462, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -576.92307692,
+          -576.92307692, -1346.15384615, -384.61538462, -384.61538462, 0, 0, 0, 0, 0, 0, 0;
+    }
+  } else
+    std::cout << "There is no stress test yet implemented for the Element " << Dune::className(fe) << std::endl;
+
+  ResultTypeMap result;
   auto resultRequirements = ResultRequirements()
                                 .insertGlobalSolution(FESolutions::displacement, displacement)
                                 .addResultRequest(ResultType::linearStress);
 
-  ResultTypeMap result;
-  fe.calculateAt(resultRequirements, Dune::FieldVector<double, dim>(0), result);
-  auto computedResult = result.getResult(ResultType::linearStress);
+  for (auto i : std::views::iota(0, nodes)) {
+    auto pos = referenceElement.position(i, dim);
+    fe.calculateAt(resultRequirements, pos, result);
+    computedStress(i, Eigen::placeholders::all) = result.getResult(ResultType::linearStress).transpose();
```

Comment:
Ah and I forgot. In this case here i would write = `computedStress.row(i) = result.getResult(ResultType::linearStress).transpose();` since i think this makes your intent more clear. Maybe even the transpose is not needed then.